src/core/symbology-ng/qgsgraduatedsymbolrendererv2.cpp

```diff
@@ -1348,6 +1315,52 @@ void QgsGraduatedSymbolRendererV2::setSourceColorRamp( QgsVectorColorRampV2* ram
   mSourceColorRamp.reset( ramp );
 }
 
+double QgsGraduatedSymbolRendererV2::minSymbolSize() const
+{
+  double min = 0;
+  for ( int i = 0; i < mRanges.count(); i++ )
+  {
+    double sz = 0;
+    if ( dynamic_cast< QgsMarkerSymbolV2 * >( mRanges[i].symbol() ) )
+      sz = dynamic_cast< QgsMarkerSymbolV2 * >( mRanges[i].symbol() )->size();
+    if ( dynamic_cast< QgsLineSymbolV2 * >( mRanges[i].symbol() ) )
+      sz = dynamic_cast< QgsLineSymbolV2 * >( mRanges[i].symbol() )->width();
+    min = i ? qMin( sz, min ) : sz;
```

Comment:
this is a bit messy - better to initialise min to DBL_MAX (and max in line 1335 to DBL_MIN) so that this can be simplified to min=qMin( sz, min)